_posts/2023-06-05-timelines-save-time-and-money.md

```diff
@@ -0,0 +1,294 @@
+---
+layout: post
+title: "It’s Time for a Time-centric Stack: Timeline Abstractions Save Time and Money"
+image: assets/images/blog/Blog_Post_Time_Centric.png
+author: Brian Godsey
+---
+
+
+
+When processing and building real-time feature values from event-based data, it
+is easy to underestimate the power of using tools that are built to understand
+how time works. Many software tools can parse and manipulate time-based data,
+but the vast majority have little more than convenience functionality for
+parsing and doing arithmetic on date-time values. On the other hand, Kaskada has
+a built-in, natural understanding of time that just works by default, which can
+reduce development time and effort, simplify deployment and debugging, and
+reduce the compute and storage resources needed to support real-time event-based
+systems.
+
+In this post, I'll discuss some costly pitfalls when building data pipelines
+with event-based data. In particular, I will share some anecdotes from my past
+experience building SQL-based pipelines, how my team spent considerable time and
+money that (in hindsight) could have been avoided, and how Kaskada specifically
+could have simplified development and reduced costs. I will get into specifics
+below, but the lessons I've learned from using SQL on event data boil down to
+this: 
+
+> Event-based data and time values have special properties and natural
+> conventions, which lead us to write the same generic logic over and over. This
+> time-centric logic—using window functions, time-JOINs, etc—can become
```

Comment:
I think this should use double dashes for the "parenthetical like phrase" to improve typesetting. `time-centric logic -- using window functions, time-JOINs, etc. -- can become".

Reply:
I'm using emdashes, but you're right that the rendering has them kinda short. I don't like double-dashes with spaces as a substitute, but I do use triple-dashes (no spaces) when emdashes aren't available.
https://www.reddit.com/r/grammar/comments/h8qw97/why_are_two_hyphens_used_in_place_of_the_em_dash/

Reply:
I believe Jekyll automatically turns three dashes into an emdash. Otherwise you need to use the HTML character code rather than the unicode emdash.